src/sentry/static/sentry/app/views/ruleEditor/index.jsx

```diff
@@ -136,6 +149,14 @@ const RuleEditor = createReactClass({
     return !!error[field];
   },
 
+  handleChange(prop, val) {
+    this.setState(state => {
+      let rule = {...state.rule};
+      rule[prop] = val;
+      return {rule};
+    });
+  },
```

Comment:
Can this be written as:

```js
this.setState(state => ({rule: {...state.rule, [prop]: val}}));
```

maybe it's too crazy, just a bit condensed.

Reply:
gonna weight in and say i think this is not better in this context.

Reply:
combines the [key] syntax with a really busy line

Reply:
I like it *shrug*

Reply:
i think it's not too crazy, but gonna leave it for now